### PR TITLE
adding git sync cron job

### PR DESCRIPTION
**Pull request by pcarney8**
#### What is this PR About?
Adding a template for a `CronJob` that will sync two git repositories

#### How do we test this?
`oc create -f cronjob-git-sync.yml`
then instantiate the template with params:
```
GIT_PRIVATE_KEY=
GIT_HOST=
GIT_PORT=
GIT_REPO=
GIT_SYNC_HOST=
GIT_SYNC_PORT=
GIT_SYNC_REPO=
SCHEDULE=*/5 * * * *
```

cc: @redhat-cop/day-in-the-life-ops

### CODE REVIEW

File: jobs/cronjob-git-sync.yml [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    successfulJobsHistoryLimit: "${{SUCCESS_JOBS_HISTORY_LIMIT}}"
+    failedJobsHistoryLimit: "${{FAILED_JOBS_HISTORY_LIMIT}}"
+    jobTemplate:
+      spec:
+        template:
+          spec:
+            volumes:
+            - name: secret-volume
+              secret:
+                secretName: ssh-key-secret
```
Comment (sabre1041) [inline jobs/cronjob-git-sync.yml]: This should be the name of the secret from the parameter

**Review by sabre1041 — DISMISSED** [review]
@pcarney8 

Validated the E2E functionality. Very cool! Added one comment. 

Also, for the SSH key, a user can also get the value by running the following to provide to the template 

```
cat <private_key> | base64
```

**Comment by etsauer** [comment]
@pcarney8 can you also add to the `jobs/README.md` with usage info?

**Comment by pcarney8** [comment]
@etsauer and @sabre1041 see updates please

**Comment by oybed** [comment]
@sabre1041 @etsauer are we good to merge now?

**Review by oybed — APPROVED** [review]
All changes have been made and will merge

**Pull request merged**